webapp-client/src/component/CategoryList.js

```diff
@@ -1,30 +1,77 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import ErrorMessage from './ErrorMessage';
 
 export default class CategoryList extends React.Component {
-  constructor(props){
+  constructor(props) {
     super(props);
-    this.addAElement = this.addAElement.bind(this);
+    this.state = {
+      hasError: false,
+      categoryList: [],
+    };
+    this.addCategoryElement = this.addCategoryElement.bind(this);
+    this.clearHasError = this.clearHasError.bind(this);
   }
 
-  addAElement(category, index){
+  async componentDidMount() {
+    const header = { 'Content-Type': 'application/json' };
+    try {
+      const response = await fetch(
+        process.env.REACT_APP_API_URL +
+          '/category/getCategories/' +
+          this.props.userID,
+        header
+      );
+      if (response.status !== 200) {
+        throw new Error(response.statusMessage);
+      }
+      const categoryList = await response.json();
+      this.setState({
+        categoryList,
+      });
+    } catch (error) {
+      this.setState({ hasError: true });
+    }
+  }
+
+  addCategoryElement(category, index) {
     return (
-      <a className="panel-block is-active" key={index} onClick={() => this.props.setContentPane(category)}>
-        {category}
+      <a
+        className="panel-block is-active"
+        key={index}
+        onClick={() => this.props.setCurrentCategory(category._id, category.title)}
+      >
+        {category.title}
       </a>
-    ); 
+    );
+  }
+
+  clearHasError() {
+    this.setState({
+      hasError: false,
+    });
   }
 
   render() {
```

Comment:
How is "selected category in the list" annotated?

Reply:
Ah i will add styling to show which category is selected